### DISCUSSION TITLE
What's the difference between a UserControl and a View?

### DISCUSSION THREAD

**Discussion by slxdy** [AvaloniaUI/Avalonia#15966] [Q&A] [answered]
What's the difference between a UserControl and a View? In which case should I use a UserControl instead of a View?

**Comment by stevemonaco** [comment, marked as answer]
"View" is more of an umbrella term than a specific type you directly use. A `TreeView` is a type of control, there are some native integrations like `AvaloniaView` for iOS, etc, but there is no `View` in Avalonia. Devs that use "View" usually mean a bunch of controls that are arranged together as a page (or a subcomponent of a page). There's also the "View" in MVVM which can be type(s) coupled to a ViewModel for visual presentation. I also use "View system" to refer to GUI frameworks in an MVVM context, but this is a personal term.

A `UserControl` is a bunch of pre-existing controls (potentially including other `UserControl`s) arranged together with the ability to use code-behind (for handling events and such). So if that's what you're doing, that's what you should use.

**Reply by slxdy** [reply]
I meant the MVVM View, but this answered my question. Thanks

**Reply by stevemonaco** [reply]
In Avalonia, you're typically only coupling `Window` or `UserControl` as your "View" to a ViewModel. You can also use `DataTemplate` as a typeless "View" without code-behind. Coupling other custom controls (ie. `TemplatedControl` or self-rendering ones) to a specific VM (instead of binding many properties) hurts the ergonomics and reusability, so I don't recommend it.
